Modules/Loadable/Segmentations/Logic/vtkSegmentEditorLogic.cxx

```diff
@@ -0,0 +1,2182 @@
+  std::string addedSegmentID = segmentationNode->GetSegmentation()->AddEmptySegment(segmentId.c_str());
+
+  // Set default terminology entry from application settings
+  vtkSegment* addedSegment = segmentationNode->GetSegmentation()->GetSegment(addedSegmentID);
+  if (addedSegment)
+  {
+    std::string defaultTerminologyEntryStr;
+    // Default terminology in the segmentation node has the highest priority
+    vtkNew<vtkSlicerTerminologyEntry> entry;
+    if (vtkSlicerTerminologiesModuleLogic::GetDefaultTerminologyEntry(segmentationNode, entry) && !entry->IsEmpty())
+    {
+      defaultTerminologyEntryStr = vtkSlicerTerminologiesModuleLogic::SerializeTerminologyEntry(entry);
+    }
+    if (defaultTerminologyEntryStr.empty())
+    {
+      defaultTerminologyEntryStr = DefaultTerminologyEntry;
+    }
+
+    if (!defaultTerminologyEntryStr.empty())
+    {
+      addedSegment->SetTerminology(defaultTerminologyEntryStr);
+    }
+  }
+
+  // Set segment status to one that is visible by current filtering criteria
+  vtkSlicerSegmentationsModuleLogic::SetSegmentStatus(addedSegment, segmentStatus);
+
+  // Select the new segment
+  if (!addedSegmentID.empty())
+  {
+    SegmentEditorNode->SetSelectedSegmentID(addedSegmentID.c_str());
+  }
+
+  // Assign the new segment the terminology of the (now second) last segment
+  if (segmentationNode->GetSegmentation()->GetNumberOfSegments() > 1)
+  {
+    vtkSegment* secondLastSegment = segmentationNode->GetSegmentation()->GetNthSegment(segmentationNode->GetSegmentation()->GetNumberOfSegments() - 2);
+    std::string repeatedTerminologyEntry = secondLastSegment->GetTerminology();
+    segmentationNode->GetSegmentation()->GetSegment(addedSegmentID)->SetTerminology(repeatedTerminologyEntry);
+  }
+  return addedSegmentID;
+}
+
+//------------------------------------------------------------------------------
+bool vtkSegmentEditorLogic::CanAddSegments() const
+{
+  // Disable adding new segments until source volume is set (or reference geometry is specified for the segmentation).
+  // This forces the user to select a source volume before start adding segments.
+  return (GetSegmentation() != nullptr) && (GetSourceVolumeNode() != nullptr) && (!GetReferenceImageGeometryString().empty());
+}
+
+//-----------------------------------------------------------------------------
+bool vtkSegmentEditorLogic::CanRedo() const
+{
+  return SegmentationHistory && SegmentationHistory->IsRestoreNextStateAvailable();
+}
+
+//------------------------------------------------------------------------------
+bool vtkSegmentEditorLogic::CanRemoveSegments() const
+{
+  // Only enable remove button if a segment is selected
+  return IsSegmentIdValid(GetCurrentSegmentID());
+}
+
+//-----------------------------------------------------------------------------
+bool vtkSegmentEditorLogic::CanTriviallyConvertSourceRepresentationToBinaryLabelMap() const
+{
+  if (!IsSegmentationNodeValid())
```

Comment:
In the remaining of the file, add missing `vtkSegmentEditorLogic::` for static calls, and missing `this->` for instance variables 🙏